lib/ansible/modules/cloud/amazon/lightsail.py

```diff
@@ -0,0 +1,405 @@
+
+    module: Ansible module object
+    client: authenticated lightsail connection object
+    instance_name: name of instance to start/stop
+    state: Target state ("running" or "stopped")
+
+    Returns a dictionary of instance information
+    about the instance started/stopped
+
+    If the instance was not able to state change,
+    "changed" will be set to False.
+
+    """
+    wait = module.params.get('wait')
+    wait_timeout = int(module.params.get('wait_timeout'))
+    wait_max = time.time() + wait_timeout
+
+    changed = False
+
+    inst = None
+    try:
+        inst = find_instance_info(client, instance_name)
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] != 'NotFoundException':
+            module.fail_json(msg='Error finding instance {0}, error: {1}'.format(instance_name, e))
+
+    # Wait for instance to exit transition state before state change
+    if wait:
+        while wait_max > time.time() and inst is not None and inst['state']['name'] in ('pending', 'stopping'):
+            try:
+                time.sleep(5)
+                inst = find_instance_info(client, instance_name)
+            except botocore.exceptions.ClientError as e:
+                time.sleep(1)
+
+    # Try state change
+    if inst is not None and inst['state']['name'] != state:
+        try:
+            if state == 'running':
+                client.start_instance(instanceName=instance_name)
+            else:
+                client.stop_instance(instanceName=instance_name)
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg='Unable to change state for instance {0}, error: {1}'.format(instance_name, e))
+        changed = True
+        # Grab current instance info
+        inst = find_instance_info(client, instance_name)
+
+    return (changed, inst)
+
+def core(module):
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    if not region:
+        module.fail_json(msg='region must be specified')
+
+    client = None
+    try:
+        client = boto3_conn(module, conn_type='client', resource='lightsail',
+                            region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except (botocore.exceptions.ClientError, botocore.exceptions.ValidationError) as e:
+        module.fail_json(msg=str(e))
+
+    changed = False
+    state = module.params['state']
+    name = module.params['name']
+
+    if state == 'absent':
+        (changed, instance_dict) = delete_instance(module, client, name)
+    elif state in ('running', 'stopped'):
+        (changed, instance_dict) = startstop_instance(module, client, name, state)
+    elif state == 'restarted':
+        (changed, instance_dict) = restart_instance(module, client, name)
+    elif state == 'present':
+        (changed, instance_dict) = create_instance(module, client, name)
+
+    module.exit_json(changed=changed, instance=instance_dict)
```

Comment:
When exiting, the current module returns keys such as:

```
"instance": {
        "arn": "arn:aws:lightsail:us-east-1:509803855674:Instance/b35a930e-bf27-43bc-9e16-82032254baab", 
        "blueprintId": "ubuntu_16_04", 
        "blueprintName": "Ubuntu", 
        "bundleId": "nano_1_0", 
        "createdAt": "2017-01-20T14:59:15.745000-05:00", 
        "hardware": {
            "cpuCount": 1, 
            "ramSizeInGb": 0.5
        }, 
        "isStaticIp": false, 
        "location": {
            "availabilityZone": "us-east-1a", 
            "regionName": "us-east-1"
        }, 
        "name": "fromansible", 
        "networking": {
.....
```

This return needs to be run through the `camel_dict_to_snake_dict` utility so that values like `isStaticIp` are returned as `is_static_ip` instead.